File: DarkOrbit.Api/Processes/ProcessEntity.cs
using System.Collections.Generic;
using DarkOrbit.Api.Utilities.Database;

namespace DarkOrbit.Api.Processes
{
    public class ProcessEntity : MongoDbEntityBase
    {
        public string Name { get; set; }

        public string Description { get; set; }

        public List<ProcessStep> Steps { get; set; }

        public List<Input> Inputs { get; set; }

        public List<Output> Outputs { get; set; }
    }
}

File: DarkOrbit.WebApp/src/app/api/models/process-entity.ts
/* tslint:disable */
import { ProcessStep } from './process-step';
import { Input } from './input';
import { Output } from './output';

export interface ProcessEntity {

  id?: string;

  name?: string;

  steps?: ProcessStep[];

  inputs?: Input[];

  outputs?: Output[];

  description?: string;

  created?: string;

  lastUpdated?: string;

  createdBy?: string;

  updatedBy?: string;

  deleted?: boolean;
}
